### PR TITLE
Feature/api gateway login

### PR DESCRIPTION
**Pull request by Ngwara**
API CA gateway implementation - Missing version module logic.
Gateway host address needs to be added to application conf

### CODE REVIEW

**Comment by RichardSmithONS** [comment]
I'm now ready for this to be merged (following any PR requested changes) as it is. There is a network issue with access to the Gateway so the login is always fake (lets you in unless your user id is MrRobot).  I think  the code could be refactored to be tidier, but I'm now satisfied it does what it should.

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala [hunk 1]
```diff
@@ -1,34 +1,132 @@
...
+  def doLogin: Action[AnyContent] = Action { implicit req =>
+
+    val formValidationResult = loginForm.bindFromRequest.data
+    (for {
+      userName <- formValidationResult.get("userName") if userName.nonEmpty
+      password <- formValidationResult.get("password") if userName.nonEmpty
+    } yield {
+    //  use fake gateway in dev or test
+      val mode: Mode = environment.mode
+ //     val realGateway: Boolean = mode match {
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala]: Do we still need this commented code?

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala [hunk 2]
```diff
@@ -1,34 +1,132 @@
...
+
+  }
+
+  /**
+    * Load login viewlet unless config says login is not required in the conf
+    * @return
+    */
+  def login: Action[AnyContent] = Action { implicit req =>
+    logger.info("loginRequired " + conf.config.loginRequired )
+    if (!conf.config.loginRequired)
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala]: Nit: avoid using negative `if` if possible as they are more difficult to follow than positive ones

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala [hunk 3]
```diff
@@ -1,34 +1,132 @@
...
+    } yield {
+    //  use fake gateway in dev or test
+      val mode: Mode = environment.mode
+ //     val realGateway: Boolean = mode match {
+  //      case Mode.Dev => false
+  //      case Mode.Test => false
+  //      case Mode.Prod => true
+  //    }
+
+      val realGateway: Boolean = false
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala]: Shouldn't this be in a config?

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala [hunk 4]
```diff
@@ -1,34 +1,132 @@
...
+          request.withHeaders("Accept" -> "application/json")
+            .withAuth(userName, password, WSAuthScheme.BASIC)
+            .withRequestTimeout(10000.millis)
+
+        val futureResponse: Future[WSResponse] = complexRequest.get()
+
+        val result = Await.result(futureResponse, 10000.millis)
+     //   logger.info("request run + result.status")
+
+        if (result.status != OK) {
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/controllers/ApplicationHomeController.scala]: How does this work? If the correct result.status with token is not 200, than what is it?

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/model/formdata/LoginCredentials.scala [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.ons.addressIndex.demoui.model.formdata
+
+/**
+  * Backing class for the AiUser data form.
+  * Requirements:
+  * <ul>
+  * <li> All fields are public,
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/model/formdata/LoginCredentials.scala]: Please remove html tags from the scaladoc

File: demo-ui/app/uk/gov/ons/addressIndex/demoui/modules/VersionModule.scala [hunk 6]
```diff
@@ -38,12 +35,15 @@ class DemoUIAddressIndexVersionModule @Inject()(
...
           resp.dataVersion
         }, 10 seconds)).getOrElse(dummyVersion)
   }
 
   val dummyVersion = "not found"
-
+  /** PLEASE Remember to modify module
+    * Add move to config -
+  */
+  val tempApiKey = "exxx-xxxxx-xxxxx-xxxxx-xxxxxx"
```
Comment (Mironor) [inline demo-ui/app/uk/gov/ons/addressIndex/demoui/modules/VersionModule.scala]: Is this still relevant? I thought we would not require api key for version request.

File: demo-ui/conf/apigw-in-d-02.crt [hunk 7]
```diff
@@ -0,0 +1,42 @@
+-----BEGIN CERTIFICATE-----
```
Comment (Mironor) [inline demo-ui/conf/apigw-in-d-02.crt]: This if fine to be public, right?

File: demo-ui/conf/application.conf [hunk 8]
```diff
@@ -1,18 +1,30 @@
 play {
   crypto.secret = "zdcqell?ol4ckQ0oP11lpkmByGJ_74EpoDqk?P3mi^v?ftdp8?Hc53RZ:LOaTeW8"
   i18n.langs = [ "en", "cy" ]
+  ws.acceptAnyCertificate=true
+  //ws.ssl {
```
Comment (Mironor) [inline demo-ui/conf/application.conf]: Commented configuration

**Comment by RichardSmithONS** [comment]
PR changes applied - working with fake Gateway only pending resolution of SSL / CF issues

**Pull request merged**